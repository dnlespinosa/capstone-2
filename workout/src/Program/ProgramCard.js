import React, { useContext, useState } from 'react';
import { Link } from 'react-router-dom';

const ProgramCard = ({ name, exercises, id }) => {

    return (
        <>
        {exercises ? 
            <div id={name} style={{backgroundColor: 'white', border: '5px solid #2a2a2a', color: '#2a2a2a', padding: '20px', margin: '10px 500px 10px 500px'}}>
            <Link to={`/browse/${name}`}>
                <h2>{name}</h2>
            </Link>
            
                {exercises?.map(e => (
                    <p key={e}>{e}</p>
                ))}
            </div> : null
        }
        
        </>
    )
}

export default ProgramCard