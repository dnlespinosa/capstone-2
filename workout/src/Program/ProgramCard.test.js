import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import ProgramCard from "./ProgramCard";
import { UserProvider } from "../testUtils";



describe('<ProgramCard />', () => {


    const exercises = [
        'Barbell Bench Press',
        'Dumbbell Alternating Single Arm Press',
        'Cable Pec Fly',
        'Cable Overhead Tricep Extension',
        'Push Up'
    ]

    // smoke test
    it("renders without crashing", function () {
        render(
            <MemoryRouter>
            <UserProvider>
                <ProgramCard />
            </UserProvider>
            </MemoryRouter>,
        );
    });
  
    // snapshot 
    it('matches snapshot', function() {
        const { asFragment } = render(
                <MemoryRouter>
                    <UserProvider>
                        <ProgramCard exercises={exercises}/>
                    </UserProvider>
                </MemoryRouter>
            )
        expect(asFragment()).toMatchSnapshot();
        })
    })

