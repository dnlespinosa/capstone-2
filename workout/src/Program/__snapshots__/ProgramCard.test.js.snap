// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgramCard /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: white; border: 5px solid #2a2a2a; color: rgb(42, 42, 42); padding: 20px; margin: 10px 500px 10px 500px;"
  >
    <a
      href="/browse/undefined"
    >
      <h2 />
    </a>
    <p>
      Barbell Bench Press
    </p>
    <p>
      Dumbbell Alternating Single Arm Press
    </p>
    <p>
      Cable Pec Fly
    </p>
    <p>
      Cable Overhead Tricep Extension
    </p>
    <p>
      Push Up
    </p>
  </div>
</DocumentFragment>
`;
